onmt/Models.py

```diff
@@ -2,13 +2,18 @@
 import torch
 import torch.nn as nn
 from torch.autograd import Variable
+from torch.nn.utils.rnn import pad_packed_sequence as unpack
+from torch.nn.utils.rnn import pack_padded_sequence as pack
+
 import onmt
 import onmt.modules
 from onmt.IO import ONMTDataset
 from onmt.modules import aeq
 from onmt.modules.Gate import ContextGateFactory
-from torch.nn.utils.rnn import pad_packed_sequence as unpack
-from torch.nn.utils.rnn import pack_padded_sequence as pack
+from onmt.Translator import DecoderState
```

Comment:
No, I don't think this file / class should know about Translator, let's keep state here.

Reply:
Yes. The `DecoderState` should be moved to some other place.  Otherwise there would be a circular dependency(`Models.py` needs the `TransformerDecoderState` in `Transformer.py`,  while `Transformer.py` need `DecoderState` in `Models.py`).